Design.MD

```diff
@@ -0,0 +1,92 @@
+# JobWorker Design Doc
+![Diagram](./images/Diagram.png "Design Approach Diagram")
+## Proposed API
+- Four endpoints: `/jobs/start`, `/jobs/stop/{id}`, `/jobs/status/{id}`, `/jobs/output/{id}`
+- If the server is running on `https://localHost:8443`, the endpoint for start would be `https://localHost:8443/jobs/start`
+- `"/jobs/"` prefix included to delineate this API from other possible endpoints, e.g. `/users/` 
+- `start` and `stop` are `POST` methods, `status` and `output` are `GET`
+
+## Component Details
+
+1. **Worker Library**
+    - Data Structures: 
+        - `Job` struct with fields like ID, status, output, cancel, pointer to exec.Cmd
+        - `jobMap`, `jobLock`: Mapping from `ID` to `Job`, and a read/write lock `sync.RWMutex` 
+    - Functions to perform tasks from the API (all public):
+        - `Start()`, `Stop()`, `Status()`, `Output()`
+    - Notes:
+        - Will use Google's UUID package for generating job ID's 
+        - When a command is started via `Start()` it will be given a context with cancel for termination in the `Stop()` function
+        - Will lock `jobMap` appropriately to prevent data races but allow concurrent reads
+2. **Server**
+    -  Data Structures: 
+        -  `User` struct with fields for `TokenID string` and `Role string`
+        -  `userMap map[string]User` map of users where key is `TokenID`
+        -  `rolePermissions map[string]map[string]bool` a map to authorize api actions based on `User.role` 
+        -  `commandPermissions map[string][]string` maps `User.role` to a list of linux commands that role can run
```

Comment:
I would avoid trying to create an allow-list of commands. Getting it right so that it can't be bypassed is a big undertaking (too big for this project), and anything less than that is kind of pointless.